src/main/java/seedu/address/logic/commands/MarkCommand.java

```diff
@@ -0,0 +1,73 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.OfficeConnectModel;
+import seedu.address.model.RepositoryModelManager;
+import seedu.address.model.task.Content;
+import seedu.address.model.task.Status;
+import seedu.address.model.task.Subject;
+import seedu.address.model.task.Task;
+
+/**
+ * Marks a task as completed
+ */
+public class MarkCommand extends Command {
+    public static final String COMMAND_WORD = "mark";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Mark the task identified by the index number as completed.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_MARK_TASK_SUCCESS = "Marked Task: %1$s";
+    private final Index taskIndex;
+
+    public MarkCommand(Index taskIndex) {
+        this.taskIndex = taskIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model, OfficeConnectModel officeConnectModel) throws CommandException {
+        requireAllNonNull(model, officeConnectModel);
+
+        RepositoryModelManager<Task> taskManager = officeConnectModel.getTaskModelManager();
+        List<Task> taskList = taskManager.getFilteredItemList();
+
+        if (taskIndex.getZeroBased() >= taskList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+
+        Task taskToMark = taskList.get(taskIndex.getZeroBased());
+        if (taskToMark.getStatus().equals(new Status(true))) {
+            throw new CommandException(Messages.MESSAGE_TASK_ALREADY_DONE);
+        }
+
+        Task markedTask = createMarkedTask(taskToMark);
+        taskManager.setItem(taskToMark, markedTask);
+
+        return new CommandResult(String.format(MESSAGE_MARK_TASK_SUCCESS, markedTask));
+    }
+
+    private static Task createMarkedTask(Task taskToMark) {
+        assert taskToMark != null;
+
+        Subject subject = taskToMark.getSubject();
+        Content content = taskToMark.getContent();
+        Status doneStatus = new Status(true);
+        return new Task(subject, content, doneStatus);
+    }
```

Comment:
The code will generate a new Task with new TaskId, which will break your taskAssignment Mapping.

Just a suggestion, Since everytime we need to create a new task object for editting, I think it would be better if we have a factory method in task class that can help us to clone a new object, in addition the factory class can enforce the isSame relationship easily.